docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+   # with the field values from the resource.
+   # mutex: RESOURCE_NAME/{{resource_id}}
+
+   # IAM_GOES_HERE
+
+   # EXAMPLES_GO_HERE
+
+   parameters:
+     - !ruby/object:Api::Type::String
+       name: 'location'
+       required: true
+       immutable: true
+       url_param_only: true
+       description: |
+         LOCATION_DESCRIPTION
+     - !ruby/object:Api::Type::String
+       name: 'resource_id'
+       required: true
+       immutable: true
+       url_param_only: true
+       description: |
+         RESOURCE_ID_DESCRIPTION
+
+   properties:
+     # Fields go here
+   ```
+
+3. Modify the template as needed to match the API resource's documented behavior.
+
+> **Note:** The template includes the most commonly-used fields. For a comprehensive reference, see [ResourceName.yaml reference ↗]({{<ref "/reference/resource-reference.md" >}}).
+{{< /tab >}}
+{{< tab "Handwritten" >}}
+> **Warning:** Handwritten resources are much more difficult to develop and maintain. New handwritten resources will only be accepted if the resource cannot be implemented using MMv1.
+
+1. Add the resource in MMv1.
+2. [Generate the beta provider]({{< ref "/get-started/generate-providers.md" >}})
+3. From the beta provider, copy the files generated for the resource to the following locations:
+   - Resource: Copy to the appropriate service folder inside [`magic-modules/mmv1/third_party/terraform/services`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/services)
+   - Documentation: [`magic-modules/mmv1/third_party/terraform/website/docs/r`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/website/docs/r)
+   - Tests: [`magic-modules/mmv1/third_party/terraform/tests`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests)
+   - Sweepers: [`magic-modules/mmv1/third_party/terraform/utils`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/utils)
+4. Modify the Go code as needed.
+   - Replace the comments at the top of the file with the following:
+     ```
+     <% autogen_exception -%>
+     ```
+   - If any of the added Go code (including any imports) is beta-only, change the file suffix to `.go.erb` and wrap the beta-only code in a version guard: `<% unless version = 'ga' -%>...<% else -%>...<% end -%>`.
+5. Register the resource in [`magic-modules/mmv1/third_party/terraform/utils/provider.go.erb`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/provider.go.erb) under "START handwritten resources"
+   - Add a version guard for any beta-only resources.
+{{< /tab >}}
+{{< /tabs >}}
+
+## Add fields
+
+{{< tabs "fields" >}}
+{{< tab "MMv1" >}}
+1. For each API field, copy the following template into the resource's `properties` attribute. Be sure to indent appropriately.
+
+```yaml
+# Supported types: String, Integer, Boolean, Double, Enum,
```

Comment:
Do we have a strategy to deal with removing these comments? I'm afraid of future PRs being 80% comments-by-volume if contributors don't delete them, making PRs much harder to review. That's fine in a single configuration file, but we expect this block to be copy-pasted dozens of times in a single file.

Reply:
Interesting question. If we're concerned that users won't automatically delete the comments themselves we could actively encourage them to clean them up - did you have something in mind beyond that?

Reply:
Nothing in mind beyond that, just trying to preempt potentially wasting 1-2 PR passes when folks have copypasted wholesale. Maybe a prefix and a linter or something to make a robot complain at 'em.

Reply:
added explicit steps re: deleting copied comments in 7e0dd7ca8